src/generators/messages/index.ts

```diff
@@ -0,0 +1,91 @@
+/*
+* This program and the accompanying materials are made available under the terms of the *
+* Eclipse Public License v2.0 which accompanies this distribution, and is available at *
+* https://www.eclipse.org/legal/epl-v20.html                                      *
+*                                                                                 *
+* SPDX-License-Identifier: EPL-2.0                                                *
+*                                                                                 *
+* Copyright Contributors to the Zowe Project.                                     *
+*                                                                                 *
+*/
+
+import { TreeItem } from "vscode";
+import { ZoweUSSNode } from "../../ZoweUSSNode";
+import { ZoweTreeNode } from "../../abstract/ZoweTreeNode";
+
+export enum MessageCategoryId {
+    dataset = "dataset",
+    datasetMember = "datasetMember"
+}
+
+export enum MessageHierarchyType {
+    generic = "generic",
+    specific = "specific"
+}
+
+export enum MessageContentType {
+    open = "open",
+    upload = "upload"
+}
+
+type CombinedNode = TreeItem | ZoweUSSNode | ZoweTreeNode;
+
+export interface IMessageItem {
+    id: MessageCategoryId;
+    type: MessageHierarchyType;
+    generic?: IMessageItem;
+    messages: { [index: string]: string };
+    check: (node: CombinedNode) => boolean;
+}
+
+const items = [
+    require("./items/dataset"),
+    require("./items/datasetMember")
+].map((item) => item.default) as IMessageItem[];
```

Comment:
This messaging structure looks kind of cool.
I'm just a bit curious about the future impact that extending this generator could have in the long run.
Do you think it will be a performance issue if we have tongs of requires for handling dynamic messages based for different nodes?

Reply:
@zFernand0 we need to use it wisely, only if string really needs to be picked up dynamically based on the node type. If we obey those rules - we should be fine.

Reply:
In the event that we decide to change this in the future, it's possible that this could become a breaking change in case extenders of the Explorer decided to rely on this dynamic mechanism (which I don't think we will be advising any time soon).

That being said, it's something to keep in mind though.
I'm cool with the changes.
Thank you for acknowledging the future implications : )